datasets/eli5/eli5.py

```diff
@@ -0,0 +1,387 @@
+    # then all answers
+    for year in range(start_year, end_year + 1):
+        start_mth = start_month if year == start_year else 1
+        end_mth = end_month if year == end_year else 12
+        months = range(start_mth, end_mth + 1)
+        for month in months:
+            if (year, month) in date_to_url_comments:
+                f_url = date_to_url_comments[(year, month)]
+                processed_comments = _download_and_select_lines(dl_manager, f_url, "comments", st_time)
+                # merge submissions and comments
+                for name in _SUB_REDDITS:
+                    merged_comments = 0
+                    for dct in processed_comments[name]:
+                        did = dct["parent_id"].split("_")[-1]
+                        if did in qa_dict[name]:
+                            merged_comments += 1
+                            qa_dict[name][did]["comments"] = qa_dict[name][did].get("comments", []) + [dct]
+            else:
+                print("Could not find comments dump file for year {:4d} month {:2d}".format(year, month))
+    # then post-process
+    res = {}
+    for name in _SUB_REDDITS:
+        qa_dct_list = [(k, _post_process(rdct, name)) for k, rdct in qa_dict[name].items() if "comments" in rdct]
+        qa_dct_list = [x for x in qa_dct_list if len(x[1]["comments"]) > 0 and name in x[1]["url"]]
+        res[name] = dict(qa_dct_list[:])
+    return res
+
+
+_DESCRIPTION = """\
+Explain Like I'm 5 long form QA dataset
+"""
+
+_CITATION = """\
+@inproceedings{DBLP:conf/acl/FanJPGWA19,
+  author    = {Angela Fan and
+               Yacine Jernite and
+               Ethan Perez and
+               David Grangier and
+               Jason Weston and
+               Michael Auli},
+  editor    = {Anna Korhonen and
+               David R. Traum and
+               Lluis Marquez},
+  title     = {{ELI5:} Long Form Question Answering},
+  booktitle = {Proceedings of the 57th Conference of the Association for Computational
+               Linguistics, {ACL} 2019, Florence, Italy, July 28- August 2, 2019,
+               Volume 1: Long Papers},
+  pages     = {3558--3567},
+  publisher = {Association for Computational Linguistics},
+  year      = {2019},
+  url       = {https://doi.org/10.18653/v1/p19-1346},
+  doi       = {10.18653/v1/p19-1346},
+}
+"""
+
+
+class Eli5Config(nlp.BuilderConfig):
+    """BuilderConfig for ExplainLikeImFive."""
+
+    def __init__(self, **kwargs):
+        """BuilderConfig for ExplainLikeImFive.
+    Args:
+      **kwargs: keyword arguments forwarded to super.
+    """
+        super(Eli5Config, self).__init__(**kwargs)
+
+
+class Eli5(nlp.GeneratorBasedBuilder):
+    """ELI5: Explain Like I'm Five long form question answering dataset."""
+
+    BUILDER_CONFIG_CLASS = Eli5Config
+    _DATA_SPLIT_URL = "https://s3.amazonaws.com/datasets.huggingface.co/nlp/datasets/eli5/reddit_data_split.json"
+
+    name = "ELI5_LFQA"
```

Comment:
```suggestion
```

We should remove this line imo. The name will be replaced by "eli5" anyway when the builder is initialized